src/hddsupport.c

```
@@ -35,11 +35,32 @@ static unsigned char hddForceUpdate = 0;
 static unsigned char hddSupportErrToasted = 0;
+// Explicit PFS lifecycle state machine. Low-level boot/config probes may transiently fail to load PS2FS/PFS;
+// they enter RECOVERING instead of publishing a user-facing error. Only a settled, verified terminal failure
+// can transition to FAILED_FINAL and emit code 222.
+static hdd_pfs_state_t hddPfsState = HDD_PFS_UNKNOWN;
+
+hdd_pfs_state_t hddGetPfsState(void)
+{
+    return hddPfsState;
+}
+
+void hddSetPfsFailedFinal(void)
```

Cppcheck (2.21.0)

[style] The function 'hddSetPfsFailedFinal' should have static linkage since it is not used outside of its translation unit.

(staticFunction)